generate_schedule.py

```diff
@@ -41,11 +41,30 @@ def clamp_to_gc(power):
         # helper function: make sure to stay within GC power limits
         return min(max(power, -gc.max_power), gc.max_power)
 
-    cars = {vid: [0, 0, 0] for vid in s.world_state.vehicles}
+    # collect and accumulate information about the entire fleet
+    total_vehicle_capacity = 0
+    total_desired_energy = 0
+    average_efficiency = 0
+    v2g_enabled = False
+    for v in s.world_state.vehicles.values():
+        total_vehicle_capacity += v.battery.capacity
+        total_desired_energy += v.desired_soc * v.battery.capacity
+        average_efficiency += v.battery.efficiency * v.battery.capacity
+        if v.vehicle_type.v2g:
+            v2g_enabled = True
+    average_efficiency /= total_vehicle_capacity
+
     flex = {
         "min": [],
         "base": [],
         "max": [],
+        "vehicles": {
+            "discharge_limit": scenario.discharge_limit,
```

Comment:
default should be 0.5 here

Reply:
Based on another PR comment, scenario.py was changed in line 42 to provide a default value for scenario.discharge_limit in case the scenario.json file does not contain a value for it.